backend/src/ml_space_lambda/app_configuration/lambda_functions.py

```diff
@@ -53,9 +73,88 @@ def update_configuration(event, context):
     return f"Successfully updated configuration for {configScope}, version {version_id}."
 
 
-@api_wrapper
-def get_configuration(event, context):
-    configScope = event["queryStringParameters"]["configScope"]
-    num_versions = int(event["queryStringParameters"].get("numVersions", 1))
+def create_instance_constraint_statement(
+    statement_id: str, actions: list[str], resources: list[str], allowed_instances: list[str]
+):
+    return {
+        "Sid": statement_id,
+        "Effect": "Allow",
+        "Action": actions,
+        "Resource": resources,
+        "Condition": {"ForAnyValue:StringEquals": {"sagemaker:InstanceTypes": allowed_instances}},
```

Comment:
Good callout. I think I'm going to leave it as-is now to not add potentially unnecessary complexity. It will be an easy change in the future if we have to refactor due to changing requirements.